### DISCUSSION TITLE
ElementPropertyReadFile

### DISCUSSION THREAD

**Discussion by lk010203** [idaholab/moose#25174] [Q&A Getting Started]
Dear All,

I am vert new to MOOSE. I was trying to run a tensor mechanics test using ElementPropertyReadFile to read a .csv file. Although I have made make -j4 in the tensor mechanics folder and I can see the  tensor_mechanics-op there, however I still can not use the ElementPropertyReadFile, could you please advise on how to enable that function. Thanks.

**Comment by GiudGiud** [comment]
Hello

This object is now named `PropertyReadFile`
is your input file somewhat old?

what error do you get?

Guillaume

**Reply by lk010203** [reply]
Hi Guillaume,

I am using this file prop_grain_read_cp_03.i from https://github.com/PengWei97/monkey/blob/main/problem/poly_cp/prop_grain_read_cp_03.i. The error I am getting is : *** ERROR ***
The following error occurred in the object "prop_read", of type "ElementPropertyReadFile".
Thanks

**Reply by GiudGiud** [reply]
what is the actual error?
you seem to have only pasted the header of the error

**Reply by lk010203** [reply]
Hi Guillame,

Here is the error:
```
*** Warning, This code is deprecated and will be removed in future versions:
/projects/moose/modules/tensor_mechanics/examples/prop_grain_read_cp_03.i:53: (Functions/tdisp/value):
The parameter 'value' is deprecated.
Use 'expression' instead.
*** Warning, This code is deprecated and will be removed in future versions:
***** Invalid Object: ElementPropertyReadFile *****
Expired on Thu Jul  1 01:00:00 2021
Update your application using the 'PropertyReadFile' object

*** Warning, This code is deprecated and will be removed in future versions:projects/moose/modules/tensor_mechanics/examples/prop_grain_read_cp_03.i:65: (UserObjects/prop_read/ngrain):
The parameter 'ngrain' is deprecated.
ngrain is deprecated, use nvoronoi instead
*** Info ***
TensorMechanics Action: selecting 'total small strain' formulation. Use `incremental = true` to select 'incremental small strain' instead.
```
Thanks

**Reply by GiudGiud** [reply]
These are not errors. They are deprecation messages. They are pretty clear on what you need to do to fix your input as well
